src/Data/ProgrammesDb/EntityRepository/PromotionRepository.php

```diff
@@ -27,8 +27,18 @@ public function findActivePromotionsByContext(array $ancestryIds, DateTimeImmuta
         $contextId = end($ancestryIds);
         $parentIds = array_slice($ancestryIds, 0, -1);
 
+        // YIKES! relatedLinks is a many-to-many join, that could result in an
```

Comment:
So promotions won't be shown twice if they have two or more related links?

Reply:
Correct. Doctrine is smart enough to collapse the two rows that the DB returns into a single promo that looks like `[pid =>'promo', 'relatedLinks' => [['pid=>'rl1'], 'pid => 'rl2']]]`

An example of where we do this elsewhere is `CoreEntityService->findByPidFull('b04v6fz7')`  where you'll get back a single programme with two genres.